### PR TITLE
CP-41450: Moved C# samples

### PR DESCRIPTION
**Pull request by kc284**
Please review only the last commit "Moved files into...", the others are transfer from the xen-api repo.

### CODE REVIEW

File: csharp/XenSdkSample/GetVariousRecords.cs [hunk 1]
```diff
@@ -0,0 +1,129 @@
+﻿/*
+ * Copyright (c) Citrix Systems, Inc.
```
Comment (katshann) [inline csharp/XenSdkSample/GetVariousRecords.cs:2]: Citrix or Cloud ?

**Review by katshann — APPROVED** [review]

File: csharp/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+```
+<host>     : an https URL pointing at the server
+<username> : a valid user on the server (e.g. root)
+<password> : the user's password
+```
+
+Run it as follows:
+
+```
+.\XenSdkSample.exe <host> <username> <password>
```
Comment (katshann) [inline csharp/README.md:18]: Python example includes port in the host URL. If this is required here also, should it be mentioned in the parameter description ?
Python example also has dummy values for host port and username in the example command. Might be nice to have these here too ?

File: csharp/XenSdkSample/GetVariousRecords.cs (hunk 1 above)
Comment (kc284) [reply csharp/XenSdkSample/GetVariousRecords.cs:2]: It's Cloud for new files, but these ones are only transferred across from a different repo, so I think we can leave them as they are.

File: csharp/README.md (hunk 2 above)
Comment (kc284) [reply csharp/README.md:18]: The port is optional - will correct,

**Pull request merged**